DataFormats/Provenance/src/ProductRegistry.cc

```diff
@@ -139,9 +139,9 @@ namespace edm {
 
   std::shared_ptr<ProductHolderIndexHelper> const&
```

Comment:
This should be changed to return ```std::shared_ptr<ProductHolderIndexHelper const>``` in order to maintain the const. Or it should return ```edm::propagate_const<std::shared_ptr<ProductHolderIndexHelper>> const```.